iothub/device/samples/PnpDeviceSamples/PnpDeviceSamples/SimpleThermostat/Program.cs

```diff
@@ -0,0 +1,153 @@
+﻿using System;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Azure.Devices.Client;
+using Microsoft.Azure.Devices.Shared;
+
+namespace SimpleThermostat
+{
+    public class Program
+    {
+        private const string DeviceConnectionString = "device_connection_string_here";
+        private const string ModelId = "dtmi:com:example:simplethermostat;1";
+
+        private static DeviceClient s_deviceClient;
+
+        public static async Task Main(string[] _)
+        {
+            await RunSampleAsync();
```

Comment:
`.ConfigureAwait(false)` is used when we want to ensure that the callback isn't forced to be continued on the original context from where the `Task` was called. 
This is essential for library code, because if the `SynchronizationContext` has limitations on the no of threads that can be used to carry out operations, we would end up with a deadlock. 

For application code, however, this is not required, since the app environment, .NET Core in this case, will have it's own `SynchronizationContext`, which will do the right thing when a callback is posted to the original context from where the task was called. So we can let it use the default behavior.

Reply:
I agree it is needed in our SDK(library) code for sure as it is safe for the consumer of the library. But this is not needed in samples (SDK board said that for track2 ADT samples). For application code, like you said it is not needed if someone is using .net core, otherwise it is needed.